### PR TITLE
[Docs] Add how to guide for customising VizroAI

### PR DESCRIPTION
**Pull request by nadijagraca**
## Description
Added how to guid for customizing VizroAI. Included brief explanation of gpt model series. 
## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:11]:
```suggestion
`VizroAI` can be initialized without any arguments, in which case it will, by default, use the OpenAI model `"gpt-3.5-turbo"` with a temperature setting of 0.
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:17]:
```suggestion
When choosing an OpenAI model, there are few things to keep in mind; the complexity of the task, cost associated with it and the desired response time.
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (stichbury) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:1]: @nadijagraca I'm not sure if we need a new page for this content. Some of it is already in the FAQ https://vizro.readthedocs.io/projects/vizro-ai/en/latest/pages/explanation/faq/ and I wonder if I should just take what you've added and move it into the relevant Q&A pair (if it's not already stated). WDYT?

Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:1]:
```suggestion
# How to customize your model
```

Comment (maxschulz-COL) [reply vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:1]: I think this might actaully be the better title - based on that I would re-read below, and change some if the things accordingly

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:5]:
```suggestion
`VizroAI` accepts a single argument named `model`. This `model` argument can either be a `str` that specifies the name of the `ChatOpenAI` model or an instantiated `ChatOpenAI` model.
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:15]:
```suggestion
## Customize by `str`
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:17]: I would add the FAQ section before here, with most of the content written there. PLease re-read to see if everything fits. Then the read know which string to but.

Then I really like you explanation about the different series, so the reader finds out more about what he might choose.

Finally, I would delete the FAQ sections alltogether - @stichbury might have a few on where to put the authors

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:23]:
```suggestion
The below example shows how to customize your model in `str` form.
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:34]:
```suggestion
## Customize by providing a `ChatOpenAI` instance
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:36]:
```suggestion
The `model` parameter in `VizroAI` can also take an initialized `ChatOpenAI` model, allowing for flexible customization by users. In the example below, we'll be customizing the `ChatOpenAI` instance.
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:38]:
```suggestion
If you prefer more creative (but potentially more unstable) responses, you can raise the temperature to a maximum of 1.
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:34]: I would add somewhere in this section on where to import from, and what we support again, with a hint that we are working on more vendors.

File: vizro-ai/mkdocs.yml
Comment (maxschulz-COL) [inline vizro-ai/mkdocs.yml:13]:
```suggestion
      - Custom models: pages/user-guides/customize-vizro-ai.md
      - Advanced charts: pages/user-guides/create-advanced-charts.md
      - Different languages: pages/user-guides/use-different-languages.md
      - Vizro-AI charts in dashboards: pages/user-guides/add-generated-chart-usecase.md
```

File: vizro-ai/changelog.d/20240425_125608_nadija_ratkusic_graca_add_how_to_guide_vizroai.md
Comment (maxschulz-COL) [inline vizro-ai/changelog.d/20240425_125608_nadija_ratkusic_graca_add_how_to_guide_vizroai.md:1]: It would be phenomenal (although a little out of scope for this PR), if you could do the same to the installation guide than for `vizro-core`: https://vizro.readthedocs.io/en/latest/pages/user-guides/install/

Ie take it out of the tutorials, put it into a guide, make sure it follows the format of guides, and that no links or referrals are breaking.

Finally we might want to start introducing sections for the guides, but I leave that up to you:
- Fundamentals (install, customize models, run methods)
- XXX (not sure what the section would be, the remaining topics are not really advanced, they are more Usage options)

**Review by maxschulz-COL — COMMENTED** [review]
LGTM! I think this is rally a step forward :)

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [reply vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:1]: @stichbury I made a slightly different suggestion below, let me know what you think. That would kill the FAQs alltogther

Comment (stichbury) [reply vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:1]: I'll take a look. @nadijagraca I can work on your branch if that's OK and move the content around as needed.

File: vizro-ai/changelog.d/20240425_125608_nadija_ratkusic_graca_add_how_to_guide_vizroai.md
Comment (stichbury) [reply vizro-ai/changelog.d/20240425_125608_nadija_ratkusic_graca_add_how_to_guide_vizroai.md:1]: @maxschulz-COL I think that is done in vizro-core (it's in `latest`) as it was in my recent PR that reorged the front page and menu labelling.

Comment (stichbury) [reply vizro-ai/changelog.d/20240425_125608_nadija_ratkusic_graca_add_how_to_guide_vizroai.md:1]: Ah, see what you mean. Sure, OK I'll do that here.

**Comment by stichbury** [comment]
@nadijagraca @maxschulz-COL I've revised the page according to @maxschulz-COL's feedback (and also made those changes to docs table of contents as suggested).

Please take a look at it now and probably best to get review approval from someone other than me as I've made this set of changes.

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:1]: It really is a guide on how to customize the model, so I would probably clearlt label it like that

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:3]:
```suggestion
## Supported models
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:4]: can we also add an overview link to the OpenAI models

File: vizro-ai/mkdocs.yml
Comment (maxschulz-COL) [inline vizro-ai/mkdocs.yml:11]:
```suggestion
          - Customize models: pages/user-guides/customize-vizro-ai.md
```
might be better? Small difference, but we are not creating custom models really, just customizing our choice

File: vizro-ai/mkdocs.yml
Comment (maxschulz-COL) [inline vizro-ai/mkdocs.yml:9]: Have we checked that this does not break the tutorial flow?

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:47]:
```suggestion
!!! example "Customize with instantiated model"
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:41]:
```suggestion
The example below customizes the `ChatOpenAI` instance further beyond the chosen default from the string instatiation. We pass the `"gpt-3.5-turbo-0125"` model from OpenAI as `model_name` for `ChatOpenAI`, which offers improved response accuracy.
```

**Review by maxschulz-COL — APPROVED** [review]
⭐ Thanks for finishing this of! And also thanks to @nadijagraca for the start! I have a few more suggestions, but I am happy where this sits now :)

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md
Comment (maxschulz-COL) [reply vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:1]: (see also now the chosen title in the TOC)

Comment (stichbury) [reply vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:1]: I went with "How to customize model usage"

File: vizro-ai/mkdocs.yml
Comment (stichbury) [reply vizro-ai/mkdocs.yml:9]: Yes, it's just a change in MKDocs because it was already in the guides folder and the self-contained installation page is separate from the tutorial, which refers to it in (it's basically step 1). So all good.

**Review by Anna-Xiong — APPROVED** [review]
thanks for the change, just have a minor comment to consider adding a summary at the beginning of initialization

**Review by stichbury — APPROVED** [review]

**Pull request merged**
